src/sockperf.cpp

```diff
@@ -3146,6 +3170,11 @@ int bringup(const int *p_daemonize) {
             }
         }
 
+        if (!rc && !fds_array_is_valid()) {
+            rc = SOCKPERF_ERR_SOCKET;
+            log_err("Sanity check failed for sockets list");
+        }
```

Comment:
please consider using rc as `SOCKPERF_ERR_FATAL` and calling `log_err` before rc setting to be consistent with other checks